### PR TITLE
Simplify Axes.scatter to better handle arguments

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR simplifies the overloaded `Axes.scatter` method as follows:

- handle an `s` (size) array as well, seaborn 0.11.1 expects these to be acceptable as positional arguments,
- renamed `c_sort{=True}` to `sortbycolor=False`, to have a better name and have the same default action as matplotlib's `Axes.scatter`

@alurban @eagoetz, this probably has implications for the summary pages, which I think rely upon the default sorting by colour.

### CODE REVIEW

**Comment by eagoetz** [comment]
@duncanmmacleod thanks for including me in the loop. I hope @alurban will let me shadow him as he would follow up on any checks needed before giving review approval. This begs the question, once Alex leaves, will you be requesting me as a reviewer for issues such as this? I'm happy to go for it, but I still feel rather green and will need time to spin up to Alex-level-expertise.

**Comment by duncanmmacleod** [comment]
@alurban, @eagoetz, can someone please approve this?

**Review by alurban — APPROVED** [review]
LGTM, thanks @duncanmmacleod

**Pull request merged**
